### PR TITLE
Add definition for connect and disconnect events

### PR DESCRIPTION
**Pull request by beaufortfrancois**
This PR adds missing definitions for `connect` and `disconnect` serial connection events. Note that it doesn't include a check for ports that have been granted access yet. I figured out it was OK to start there and iterate with the definition of `getPorts()` in an upcoming PR.

@reillyeon WDYT?

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -94,8 +94,14 @@
 
   ### <dfn>onconnect</dfn> attribute
 
+  The {{Serial/onconnect}} attribute is an Event handler IDL attribute for the
```
Comment (reillyeon) [inline index.html]: Is there a reference we can use for "event handler IDL attribute"? Why is "Event" capitalized?

File: index.html [hunk 2]
```diff
@@ -379,6 +385,58 @@
...
+      // Remove |event.port| from the UI.
+    });
+  </pre>
+
+  ### Serial port connection
+
+  When the UA detects a new serial port connected to the host it MUST
+  perform the following steps for each script execution environment:
+
+  1. Let |port| be the {{SerialPort}} object representing the new serial port
```
Comment (reillyeon) [inline index.html]: As in `requestPort()` we haven't been very specific about what it means to have access to a port but we could at least reference the "Security considerations" section as we do there.

Comment (reillyeon) [inline index.html]: Specify the types for variables: `|port:SerialPort|` and `|event:SerialConnectionEvent|`

**Review by reillyeon — CHANGES_REQUESTED** [review]

File: index.html (hunk 1 above)
Comment (beaufortfrancois) [reply index.html]: Fixed in https://github.com/reillyeon/serial/pull/5/commits/099e40309d28cd737c74f9787eef65289a5029ba

File: index.html (hunk 2 above)
Comment (beaufortfrancois) [reply index.html]: Fixed in https://github.com/reillyeon/serial/pull/5/commits/4bd2b1f5c8789fc00898c9c388b8f1f1116c3340

Comment (beaufortfrancois) [reply index.html]: Fixed in https://github.com/reillyeon/serial/pull/5/commits/b507ba92a36a753af6cd290c2a7e45e031867377

**Pull request closed without merging**